src/mixins/filing-template-mixin.ts

```diff
@@ -216,21 +211,16 @@ export default class FilingTemplateMixin extends Vue {
     this.setNameRequest(filing.incorporationApplication.nameRequest)
 
     // Set Name Translations
-    if (filing.incorporationApplication.nameTranslations instanceof Array) {
-      // If it's an array that means it's a draft which is saved from edit-ui by staff.
-      this.setNameTranslations(filing.incorporationApplication.nameTranslations)
-    } else {
-      // If it's an object that means it's an initial draft created from filing-ui and has a structure of an IA.
-      this.setNameTranslations(
-        filing.incorporationApplication.nameTranslations?.new?.map(x => {
-          return {
-            value: x,
-            oldValue: null,
-            action: null
-          }
-        })
-      )
-    }
+    this.setNameTranslations(
+      filing.incorporationApplication.nameTranslations?.map(x => {
```

Comment:
The optional chaining means this value could be "undefined", so you probably want to add a `|| []` in here.

Reply:
Hmm, actually, it looks like this "map" function is converting `{ id, name, oldName, action }` into `{ id, name, oldName, action }` -- in others words, is it really needed?

Reply:
For the first time (when we initiate a correction) the oldName and action props are not available in the api response, which creates an issue of not having these props in store. Due to missing props change event was not triggering if the action value is changed (at the time of Delete there is no other prop change except action). To handle this scenario I had to keep this structure.

Reply:
OK, sounds fine. Could you please add a comment similar to your response, for when another developer wonders about this code? Thanks!